model/task/task_test.go

```diff
@@ -3523,6 +3523,10 @@ func TestArchiveMany(t *testing.T) {
 		assert.True(t, task.Archived)
 		assert.Equal(t, 0, task.Execution)
 	}
+
+	// We shouldn't error if we try archiving again, in case we got stuck part way.
+	err = ArchiveMany(ctx, tasks)
+	assert.NoError(t, err)
```

Comment:
Would it be worth checking that on top of just not erroring, archiving a second time is a no-op and doesn't somehow archive the already-archived tasks again?

Same thing with the test below.

Reply:
sure!